src/edu/washington/escience/myria/storage/TupleUtils.java

```diff
@@ -449,8 +475,41 @@ public static boolean tupleEquals(
             return false;
           }
           break;
+        case BLOB_TYPE:
+          if (!table1.getBlob(compareColumns[i], row1).equals(table2.getBlob(i, index))) {
+            return false;
+          }
+          break;
       }
     }
     return true;
   }
+  /**
+   * batch size for tuple batch depending on schema.
+   * @param schema of tuplebatch
+   * @return batchsize.
+   */
+  public static int getBatchSize(Schema schema) {
```

Comment:
A simpler implementation is: `if (schema.getColumbTypes().indexOf(Type.BLOB_TYPE) >= 0) return 1`